crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -715,6 +715,14 @@ pub fn comparator_operand_type_rules(
             }
             Err(format!("Unsupported types for comparison: {} and {}", name_a, name_b))
         }
+        (String(x_size), String(y_size)) => {
+            if x_size != y_size {
```

Comment:
This is a bug fixed in the numeric generics PR. Since these are types we should unify them instead of checking `!=`. Reason being that unification will correctly handle type variables.

Reply:
Made the switch. I wasn't sure if I should put the change from your numeric generics PR in this one, but I put it in anyway as it is a small change.